swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
+
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except Exception:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = []
+
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        
+        return items_by_category
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+
+
+        else:
+            self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+            return True
+
+    
+    def get_best_by_category(self, category):
+        
+        best_condition_so_far = 0
+        
+        list_of_items_by_category = self.get_by_category(category)
+        if list_of_items_by_category == None:
+            return None
+        for item in list_of_items_by_category:
+            if item.condition > best_condition_so_far:
+                best_condition_so_far = item.condition
+        for item in list_of_items_by_category:
+            if item.condition == best_condition_so_far:
+                return item
```

Comment:
We could do these two loops "at once". We could have a second variable that tracks the best item found so far (initialize it to `None`) and each time we update the condition, also update the item. Then at the end of the loop, our second variable would already have the best item and we could return it immediately.

Note that from a big O perspective, what you've written here and what I described are both O(n) with respect to the number of items. Your approach would translate more directly into an approach using `max`.